### PR TITLE
Fix permissions issues using gtar on mac

### PR DESCRIPTION
**Pull request by dhadka**
Fixes https://github.com/actions/cache/issues/527

### CODE REVIEW

File: packages/cache/src/internal/tar.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ async function getTarPath(
     case 'darwin': {
       const gnuTar = await io.which('gtar', false)
       if (gnuTar) {
+        // fix permission denied errors when extracting BSD tar archive with GNU tar - https://github.com/actions/cache/issues/527
+        args.push('--delay-directory-restore')
```
Comment (thboop) [inline packages/cache/src/internal/tar.ts:30]: Changes look good, however we have a test that verifies what args these are called with you will need to modify.

**Comment by konradpabjan** [comment]
Nit: Could you also update `releases.md` with a sentence or two

File: packages/cache/src/internal/tar.ts (hunk 1 above)
Comment (dhadka) [reply packages/cache/src/internal/tar.ts:30]: This does unfortunately append this arg to all invocations of `tar`, including list and create, but appears to be silently ignored in those cases.

Comment (dhadka) [reply packages/cache/src/internal/tar.ts:30]: Tests are fixed

**Review by konradpabjan — APPROVED** [review]

File: packages/cache/RELEASES.md [hunk 2]
```diff
@@ -36,4 +36,7 @@
 
 ### 1.0.6
 - Make caching more verbose [#650](https://github.com/actions/toolkit/pull/650)
-- Use GNU tar on macOS if available [#701](https://github.com/actions/toolkit/pull/701)
\ No newline at end of file
+- Use GNU tar on macOS if available [#701](https://github.com/actions/toolkit/pull/701)
+
+### 1.0.7
+- Fixes permissions issue extracting archives with GNU tar on macOS ([issue](https://github.com/actions/cache/issues/527))
```
Comment (thboop) [inline packages/cache/RELEASES.md:42]: NIT: Fixes permissions issue **when** extracting archives with GNU tar on macOS

**Review by thboop — APPROVED** [review]
LGTM

**Pull request merged**